### PR TITLE
Fix greedy read

### PR DESCRIPTION
**Pull request by netixx**
# Description

Proposed fix for issue https://github.com/scrapli/scrapli_netconf/issues/122 (more details in the issue).

TLDR; scrapli was consuming too much input, which could lead to chunks of the response being discarded.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested on Cisco IOS-XR and Juniper Junos-EVO lab devices.
Ran unit test.

# Checklist:

- [X] My code follows the style guidelines of this project (no GitHub actions complaints! run `make lint` before
 committing!)
- [X] I have commented my code, pydocstyle and darglint are happy, docstrings are in google docstring format, and all
 docstrings include a summary, args, returns and raises fields (even if N/A)
- [ ] I have added tests that prove my fix is effective or that my feature works, if adding "functional" tests please
 note if there are any considerations for the vrnetlab setup
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by carlmontanari** [comment]
going to close (for now?!) and continue working on this over in scrapli netconf!
